evalml/data_checks/target_leakage_data_check.py

```diff
@@ -58,19 +58,18 @@ def validate(self, X, y):
             "warnings": [],
             "errors": []
         }
-
         X = _convert_to_woodwork_structure(X)
         y = _convert_to_woodwork_structure(y)
-        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+        if y.logical_type not in numeric_and_boolean_ww:
+            return messages
+        X_num = X.select(include=numeric_and_boolean_ww)
```

Comment:
Is there a reason to include boolean too? Perhaps this should just be `X.select('numeric')`

Reply:
I guess this should just be based on whether or not we want to / it makes sense to check for label leakage if the target is boolean :)